modules/db/src/test/java/org/mule/module/db/integration/xa/AbstractXaTransactionalTestCase.java

```diff
@@ -65,37 +68,47 @@ public void commitsChanges() throws Exception
     @Test
     public void rollbacksChanges() throws Exception
     {
-        LocalMuleClient client = muleContext.getClient();
-        MuleMessage response = client.send("vm://testRollback", TEST_MESSAGE, null);
-
-        assertTrue(response.getPayload() instanceof NullPayload);
-
-        List<Map<String, String>> result = selectData("select * from PLANET where POSITION=4", getDefaultDataSource());
-        assertRecords(result, new Record(new Field("NAME", MARS.getName()), new Field("POSITION", 4)));
+        try
+        {
+            runTransactionalFlow("jdbcRollback", ACTION_ALWAYS_BEGIN, new XaTransactionFactory());
+            fail("Expected 'java.lang.IllegalStateException'");
```

Comment:
Why is expecting IllegalStateException and then catches only ComponentException?